### DISCUSSION TITLE
Error: Directory not empty @ dir_s_rmdir - /private/tmp/d20210304-15055-61b9qs

### DISCUSSION THREAD

**Discussion by georgew509** [Homebrew/discussions#988] [Everyday usage]
I am trying to install php@7.4 on Intel Mac after a clean install of brew.

I keep getting these errors: "Error: Directory not empty @ dir_s_rmdir - /private/tmp/..."

For example:

==> Installing php@7.4 dependency: rtmpdump
==> Pouring rtmpdump-2.4+20151223_1.big_sur.bottle.tar.gz
Error: Directory not empty @ dir_s_rmdir - /private/tmp/d20210304-15055-61b9qs

$ ls -l /private/tmp/d20210304-15055-61b9qs
total 0
drwxrwxrwx  3 george  wheel  96 Mar  4 11:28 rtmpdump

I am stumped. Any ideas about what's going on?

$ brew --version
Homebrew 3.0.4
Homebrew/homebrew-core (git revision 655976; last commit 2021-03-04)

**Comment by SMillerDev** [comment]
what do `brew doctor` and `brew config` say?

**Reply by georgew509** [reply]
$ brew doctor
Your system is ready to brew.

$ brew config
HOMEBREW_VERSION: 3.0.4
ORIGIN: https://github.com/Homebrew/brew
HEAD: ee52b19179d056f3dd2fdf9c52214d3d51dbb9d9
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 6559768c37c501716601f18362d1abf55479fe29
Core tap last commit: 3 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit haswell
Clang: 12.0 build 1200
Git: 2.24.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.2-x86_64
CLT: 12.4.0.0.1.1610135815
Xcode: 12.4

**Comment by georgew509** [comment]
I kept on manually removing /private/tmp/d* directories and made progress, but I now I am getting other errors:

==> Installing php@7.4 dependency: oniguruma
==> Pouring oniguruma-6.9.6.big_sur.bottle.tar.gz
Error: Failed to read Mach-O binary: /usr/local/Cellar/oniguruma/6.9.6/COPYING

==> Installing php@7.4 dependency: tidy-html5
==> Pouring tidy-html5-5.6.0.big_sur.bottle.tar.gz
Error: Interrupted system call @ dir_initialize - /private/tmp/d20210304-20108-19ts4u2/tidy-html5

**Reply by georgew509** [reply]
Finally, after multiple brew install attempts, I was able to install it:

$ /usr/local/opt/php@7.4/bin/php -v
PHP 7.4.15 (cli) (built: Feb 27 2021 18:23:50) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.15, Copyright (c), by Zend Technologies

I had to manually clean /private/tmp/* (sudo rm -rf /private/tmp/d*) and rerun brew install php@7.4. Each time, I got a little further until it was finally installed.
